package/com.unity.formats.usd/Runtime/Scripts/IO/SanitizedSamples.cs

```diff
@@ -382,17 +381,24 @@ internal void Triangulate(bool changeHandedness)
         internal bool ShouldUnweldVertices(bool bindMaterials)
         {
             // If any primvar is face varying (1 value per vertex) or uniform (1 value per face), all  primvars + mesh attributes will have to be converted to face varying
+            bool shouldUnweldColors = colors != null && (colors.GetInterpolationToken() == UsdGeomTokens.uniform || colors.GetInterpolationToken() == UsdGeomTokens.faceVarying);
             // TODO: expose interpolation for standard mesh attributes (normals, tangents)
             bool shouldUnweldNormals = normals != null && (normals.Length == originalFaceVertexCounts.Length || normals.Length > points.Length);
-            bool shouldUnweldColors = colors != null && (colors.GetInterpolationToken() == UsdGeomTokens.uniform || colors.GetInterpolationToken() == UsdGeomTokens.faceVarying);
             bool shouldUnweldTangents = tangents != null && (tangents.Length == originalFaceVertexCounts.Length || tangents.Length > points.Length);
-            bool isTextureMappingFaceVarying = st.GetInterpolationToken() == UsdGeomTokens.faceVarying ||
-                                               uv.GetInterpolationToken() == UsdGeomTokens.faceVarying ||
-                                               uv2.GetInterpolationToken() == UsdGeomTokens.faceVarying ||
-                                               uv3.GetInterpolationToken() == UsdGeomTokens.faceVarying ||
-                                               uv4.GetInterpolationToken() == UsdGeomTokens.faceVarying;
 
-            return arePrimvarsFaceVarying || shouldUnweldNormals || shouldUnweldColors || shouldUnweldTangents || (bindMaterials && isTextureMappingFaceVarying);
+            return arePrimvarsFaceVarying || shouldUnweldNormals || shouldUnweldColors || shouldUnweldTangents || (bindMaterials && AreAnyArbitraryPrimvarsFaceVarying());
+        }
+
+        internal bool AreAnyArbitraryPrimvarsFaceVarying()
```

Comment:
Is Primvars 1 word or 2 words?
if so I think it should be "AreAnyArbitraryPrimVarsFaceVarying"